test/integration/integration-test.js

```diff
@@ -0,0 +1,193 @@
+
+const wallet = require('./wallet');
+
+const settingsSpecification = require('../fixtures/settings-specification');
+const trustlineSpecification = require('../fixtures/trustline-specification');
+const payments = require('../fixtures/payments');
+
+const RippleAPI = require('../../src').RippleAPI;
+
+
+const TIMEOUT = 20000;   // how long before each test case times out
+const INTERVAL = 2000;   // how long to wait between checks for validated ledger
+
+
+function verifyTransaction(testcase, hash, type, options, done) {
+  const currentOptions = {
+    minLedgerVersion: options.minLedgerVersion,
+    maxLedgerVersion: testcase.api.getLedgerVersion()
+  };
+  testcase.api.getTransaction(hash, currentOptions, (err, data) => {
+    if (err instanceof errors.NotFoundError
+          && testcase.api.getLedgerVersion() <= options.maxLedgerVersion) {
+      console.log('NOT VALIDATED YET...');
+      setTimeout(_.partial(verifyTransaction, testcase, hash, type, options, done), INTERVAL);
+      return;
+    } else if (err) {
+      done(err);
+      return;
+    }
+    assert(data && data.outcome);
+    assert.strictEqual(data.type, type);
+    assert.strictEqual(data.address, wallet.getAddress());
+    assert.strictEqual(data.outcome.result, 'tesSUCCESS');
+    done();
+  });
+}
+
+function testTransaction(testcase, txData, type, currentLedgerVersion, done) {
+  const signedData = testcase.api.sign(txData, wallet.getSecret());
+  console.log('PREPARED...');
+  testcase.api.submit(signedData.signedTransaction, (_err, data) => {
+    console.log('SUBMITTED...');
+    if (_err) {
+      done(_err);
+      return;
+    }
+    assert.strictEqual(data.engine_result, 'tesSUCCESS');
+    const options = {
+      minLedgerVersion: currentLedgerVersion,
+      maxLedgerVersion: txData.LastLedgerSequence
+    };
+    setTimeout(_.partial(verifyTransaction, testcase, signedData.id, type, options, done), INTERVAL);
+  });
+}
+
+function verifyResult(transactionType, transaction) {
+  assert(transaction);
+  assert.strictEqual(transaction.Account, wallet.getAddress());
+  assert.strictEqual(transaction.TransactionType, transactionType);
+}
+
+
+function setup(done) {
+  this.api = new RippleAPI({servers: ['wss://s1.ripple.com:443']});
+  // this.api = new RippleAPI({servers: ['wss://s-east.ripple.com:443']});
+  this.api.connect(() => {
+    this.api.remote.getServer().once('ledger_closed', () => done());
+  });
+}
+
+function teardown(done) {
+  this.api.remote.disconnect(() => {
+    // this.mockRippled.close();
+    setImmediate(done);
+  });
+}
+
+describe.skip('integration tests', function() {
+  const instructions = {maxLedgerVersionOffset: 100};
+  this.timeout(TIMEOUT);
+
+  beforeEach(setup);
+  afterEach(teardown);
+
+  it('settings', function(done) {
+    const currentLedgerVersion = this.api.getLedgerVersion();
+    this.api.prepareSettings(wallet.getAddress(), settingsSpecification, instructions, (err, txData) => {
+      if (err) {
```

Comment:
Oh right, because the subsequent function is also async. I guess this is actually a situation for `async.waterfall` and `_.partial`. Do you think it would make it more readable to use `async.waterfall`?